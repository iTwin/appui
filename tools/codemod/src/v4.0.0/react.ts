/*---------------------------------------------------------------------------------------------
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
import type { API, FileInfo, Options } from "jscodeshift";
import { useExtensions } from "../utils/Extensions";

export default function transformer(
  file: FileInfo,
  api: API,
  options: Options
) {
  const j = api.jscodeshift;
  useExtensions(j);

  const root = j(file.source);

  root.rename(
    "@itwin/appui-react:CommonWidgetProps",
    "@itwin/appui-react:Widget"
  );
  root.rename("@itwin/appui-react:WidgetProps", "@itwin/appui-react:Widget");

  return root.toSource(options.printOptions);
}
