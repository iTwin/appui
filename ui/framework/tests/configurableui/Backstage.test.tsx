/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
import * as React from "react";
import { mount, shallow } from "enzyme";
import { expect } from "chai";
import * as sinon from "sinon";
import {
  Backstage,
  CommandLaunchBackstageItem,
  FrontstageLaunchBackstageItem,
  SeparatorBackstageItem,
  TaskLaunchBackstageItem,
  FrontstageManager,
  FrontstageActivatedEventArgs,
  ConfigurableUiManager,
  TaskPropsList,
  WorkflowPropsList,
} from "../../src/index";
import TestUtils from "../TestUtils";
import NZ_BackstageItem from "@bentley/ui-ninezone/lib/backstage/Item";

describe("Backstage", () => {

  before(async () => {
    await TestUtils.initializeUiFramework();

    FrontstageManager.setActiveFrontstageDef(undefined);
  });

  describe("<Backstage />", () => {
    it("should render - isVisible", () => {
      mount(<Backstage isVisible={true} />);
    });

    it("should render - !isVisible", () => {
      mount(<Backstage isVisible={false} />);
    });

    it("renders correctly - isVisible", () => {
      shallow(<Backstage isVisible={true} />).should.matchSnapshot();
    });

    it("renders correctly - !isVisible", () => {
      shallow(<Backstage isVisible={false} />).should.matchSnapshot();
    });

    it("with child items", () => {
      const commandHandler = { execute: () => { } };
      shallow(
        <Backstage isVisible={true}>
          <CommandLaunchBackstageItem commandId="my-command-id" labelKey="UiFramework:tests.label" iconClass="icon-placeholder" commandHandler={commandHandler} />
          <SeparatorBackstageItem />
          <FrontstageLaunchBackstageItem frontstageId="Test1" labelKey="UiFramework:tests.label" iconClass="icon-placeholder" />
          <SeparatorBackstageItem />
          <TaskLaunchBackstageItem taskId="Task1" workflowId="ExampleWorkflow" labelKey="UiFramework:tests.label" iconClass="icon-placeholder" />
        </Backstage>,
      ).should.matchSnapshot();
    });

    it("SeparatorBackstageItem should render", () => {
      mount(<SeparatorBackstageItem />);
    });

    it("SeparatorBackstageItem renders correctly", () => {
      shallow(<SeparatorBackstageItem />).should.matchSnapshot();
    });

    it("CommandLaunchBackstageItem should render & execute", () => {
      const spyMethod = sinon.stub();
      const commandHandler = { execute: spyMethod };
      const wrapper = mount(<CommandLaunchBackstageItem commandId="my-command-id" labelKey="UiFramework:tests.label" subtitleId="UiFramework:tests.subtitle" iconClass="icon-placeholder" commandHandler={commandHandler} />);
      const backstageItem = wrapper.find(NZ_BackstageItem);
      backstageItem.find(".nz-backstage-item").simulate("click");
      expect(spyMethod.calledOnce).to.be.true;
    });

    it("CommandLaunchBackstageItem renders correctly", () => {
      const commandHandler = { execute: () => { } };
      shallow(<CommandLaunchBackstageItem commandId="my-command-id" labelKey="UiFramework:tests.label" iconClass="icon-placeholder" commandHandler={commandHandler} />).should.matchSnapshot();
    });

    it("FrontstageLaunchBackstageItem should render & execute", () => {
      const spyMethod = sinon.stub();
      const frontstageProps = {
        id: "Test1",
        defaultToolId: "PlaceLine",
        defaultLayout: "TwoHalvesVertical",
        contentGroup: "TestContentGroup1",
        defaultContentId: "TestContent1",
      };
      ConfigurableUiManager.loadFrontstage(frontstageProps);
      const remove = FrontstageManager.onFrontstageActivatedEvent.addListener((_args: FrontstageActivatedEventArgs) => spyMethod());
      const wrapper = mount(<FrontstageLaunchBackstageItem frontstageId="Test1" labelKey="UiFramework:tests.label" iconClass="icon-placeholder" />);
      const backstageItem = wrapper.find(NZ_BackstageItem);
      backstageItem.find(".nz-backstage-item").simulate("click");
      setImmediate(() => {
        expect(spyMethod.calledOnce).to.be.true;
        remove();
      });
    });

    it("FrontstageLaunchBackstageItem renders correctly", () => {
      shallow(<FrontstageLaunchBackstageItem frontstageId="Test1" labelKey="UiFramework:tests.label" iconClass="icon-placeholder" />).should.matchSnapshot();
    });

    it("TaskLaunchBackstageItem should render & execute", () => {
      const frontstageProps = {
        id: "Test1",
        defaultToolId: "PlaceLine",
        defaultLayout: "TwoHalvesVertical",
        contentGroup: "TestContentGroup1",
        defaultContentId: "TestContent1",
      };

      ConfigurableUiManager.loadFrontstage(frontstageProps);

      const taskPropsList: TaskPropsList = {
        tasks: [
          {
            id: "Task1",
            primaryStageId: "Test1",
            iconClass: "icon-placeholder",
            labelKey: "SampleApp:backstage.task1",
          },
        ],
      };

      ConfigurableUiManager.loadTasks(taskPropsList);

      // Test Workflows
      const workflowPropsList: WorkflowPropsList = {
        defaultWorkflowId: "default-workflow",
        taskPicker: {
          classid: "taskpicker-class",
          iconClass: "taskpicker-icon",
          labelKey: "taskpicker-label",
        },
        workflows: [
          {
            id: "ExampleWorkflow",
            iconClass: "icon-placeholder",
            labelKey: "Protogist:Test.my-label",
            defaultTaskId: "task1",
            tasks: ["Task1"],
          },
        ],
      };

      ConfigurableUiManager.loadWorkflows(workflowPropsList);

      const spyMethod = sinon.stub();
      const remove = FrontstageManager.onFrontstageActivatedEvent.addListener((_args: FrontstageActivatedEventArgs) => spyMethod());
      const wrapper = mount(<TaskLaunchBackstageItem taskId="Task1" workflowId="ExampleWorkflow" labelKey="UiFramework:tests.label" iconClass="icon-placeholder" />);
      const backstageItem = wrapper.find(NZ_BackstageItem);
      backstageItem.find(".nz-backstage-item").simulate("click");
      setImmediate(() => {
        expect(spyMethod.calledOnce).to.be.true;
        remove();
      });
    });

    it("TaskLaunchBackstageItem renders correctly", () => {
      shallow(<TaskLaunchBackstageItem taskId="Task1" workflowId="ExampleWorkflow" labelKey="UiFramework:tests.label" iconClass="icon-placeholder" />).should.matchSnapshot();
    });

  });
});
