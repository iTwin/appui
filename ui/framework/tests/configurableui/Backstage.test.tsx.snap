// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backstage <Backstage /> CommandLaunchBackstageItem renders correctly 1`] = `
<BackstageItem
  icon={
    <Icon
      iconInfo={
        Object {
          "iconClass": "icon-placeholder",
          "iconElement": undefined,
        }
      }
    />
  }
  key="my-command-id"
  label="tests.label"
  onClick={[Function]}
/>
`;

exports[`Backstage <Backstage /> FrontstageLaunchBackstageItem renders correctly 1`] = `
<BackstageItem
  icon={
    <Icon
      iconInfo={
        Object {
          "iconClass": "icon-placeholder",
          "iconElement": undefined,
        }
      }
    />
  }
  isActive={true}
  isDisabled={false}
  key="Test1"
  label="tests.label"
  onClick={[Function]}
/>
`;

exports[`Backstage <Backstage /> SeparatorBackstageItem renders correctly 1`] = `
<BackstageSeparator
  key="NaN"
/>
`;

exports[`Backstage <Backstage /> TaskLaunchBackstageItem renders correctly 1`] = `
<BackstageItem
  icon={
    <Icon
      iconInfo={
        Object {
          "iconClass": "icon-placeholder",
          "iconElement": undefined,
        }
      }
    />
  }
  key="ExampleWorkflow:Task1"
  label="tests.label"
  onClick={[Function]}
/>
`;

exports[`Backstage <Backstage /> renders correctly - !isVisible 1`] = `
<Fragment>
  <Backstage
    isOpen={false}
    onClose={[Function]}
    showOverlay={true}
  />
</Fragment>
`;

exports[`Backstage <Backstage /> renders correctly - isVisible 1`] = `
<Fragment>
  <Backstage
    isOpen={true}
    onClose={[Function]}
    showOverlay={true}
  />
</Fragment>
`;

exports[`Backstage <Backstage /> with child items 1`] = `
<Fragment>
  <Backstage
    isOpen={true}
    items={
      Array [
        <CommandLaunchBackstageItem
          commandHandler={
            Object {
              "execute": [Function],
            }
          }
          commandId="my-command-id"
          iconClass="icon-placeholder"
          labelKey="UiFramework:tests.label"
        />,
        <SeparatorBackstageItem />,
        <FrontstageLaunchBackstageItem
          frontstageId="Test1"
          iconClass="icon-placeholder"
          labelKey="UiFramework:tests.label"
        />,
        <SeparatorBackstageItem />,
        <TaskLaunchBackstageItem
          iconClass="icon-placeholder"
          labelKey="UiFramework:tests.label"
          taskId="Task1"
          workflowId="ExampleWorkflow"
        />,
      ]
    }
    onClose={[Function]}
    showOverlay={true}
  />
</Fragment>
`;
